website/static/js/pages/base-page.js

```diff
@@ -92,6 +93,101 @@ $(document).on('click', '.panel-heading', function(){
     }
 });
 
+//choose which emails/users to add to an account and which to deny
+function confirmEmails(emailsToAdd) {
+    if (emailsToAdd.length > 0) {
+        var confirmedEmailURL = window.contextVars.confirmedEmailURL;
+        var email = emailsToAdd[0];
+        var title;
+        var requestMessage;
+        var confirmMessage;
+        var nopeMessage;
+        if (email.user_merge) {
+            title = 'Merge account';
+            requestMessage = 'Would you like to merge \<b>' + email.address + '\</b> into your account?  ' +
+                'This action is irreversable.';
+            confirmMessage = '\<b>' + email.address + '\</b> has been merged into your account.';
+            nopeMessage = 'You have chosen to not merge \<b>' + email.address + '\</b>  into your account. ' +
+                'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+        else {
+            title = 'Add email';
+            requestMessage = 'Would you like to add \<b>' + email.address + '\</b> to your account?';
+            confirmMessage = '\<b>' + email.address + '\</b> has been added into your account.';
+            nopeMessage = 'You have chosen not to add \<b>' + email.address + '\</b> to your account.' +
+                'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+
+        var confirmFailMessage = 'There are a problem adding \<b>' + email.address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        var cancelFailMessage = 'There are a problem removing \<b>' + email.address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        bootbox.dialog({
+            title: title,
+            message: requestMessage,
+            onEscape: function () {
+            },
+            backdrop: true,
+            closeButton: true,
+            buttons: {
+                confirm: {
+                    label: 'Add email',
+                    className: 'btn-success',
+                    callback: function () {
+                        $osf.putJSON(
+                            confirmedEmailURL,
+                            email
+                        ).done(function () {
+                            $osf.growl('Success', confirmMessage, 'success', 3000);
+                            confirmEmails(emailsToAdd.slice(1));
```

Comment:
There's an additional call to `confirmEmails` a few lines down; won't this result in duplicate confirmation steps?